docs/design/client-requests.md

```diff
@@ -4,61 +4,73 @@ sidebar_position: 3
-[operation events](../reference/operations/index.md) sent to the cluster in a single message.
-
-- All events within a request batch share the same operation type.
-- The cluster commits an entire batch at once, each event in series.
-- The cluster returns a single reply for each unique request it commits.
-- The cluster's reply contains results corresponding to each event in the request.
-- Unless [linked](../reference/transfers.md#flagslinked), events within a request
-  [succeed or fail](../reference/operations/create_transfers.md#result) independently.
+A _request_ is a [batch](#batching-events) of one or more [operation
+events](../reference/operations/index.md) sent to the cluster in a single message.
+
+- All events within a request batch must be of the same
+  [operation](../reference/operations/index.md) type. You cannot, for example, create accounts and
+  transfers in the same request.
+- The cluster commits an entire batch at once. Events are applied in series, such that successive
+  events observe the effects of previous ones and event timestamps are [totally
+  ordered](time.md#timestamps-are-totally-ordered).
+- Events within a request [succeed or fail](../reference/operations/create_transfers.md#result)
+  independently unless they are explicitly [linked](../reference/transfers.md#flagslinked).
+- The cluster returns a single reply for each unique request it commits. The reply contains a
+  [result](../reference/operations/create_transfers.md#result) for each event in the request.
 
 ## Batching Events
 
-To achieve high throughput, TigerBeetle amortizes the overhead of consensus and I/O by
-batching many operation events in each request. For the best performance, each request should batch
-as many events as possible. Typically this means funneling events through fewer client instances
-(e.g. a single client instance per process).
+To achieve high throughput, TigerBeetle amortizes the overhead of consensus and I/O by batching many
+operation events in each request. 
+
+For the best performance, each request should batch as many events as possible. Typically this means
+funneling events through fewer client instances (e.g. a single client instance per process).
 
 The maximum number of events per batch depends on the maximum message size
-(`config.message_size_max`) and the operation type.
-(TODO: Expose batch size in the client instead).
+(`config.message_size_max`, specified in bytes) and the operation type. (TODO: Expose batch size in
+the client instead).
 
 In the default configuration, the batch sizes are:
 
-| Operation               | Batch Size |
-| ----------------------- | ---------: |
-| `lookup_accounts`       | 8190       |
-| `lookup_transfers`      | 8190       |
-| `create_accounts`       | 8190       |
-| `create_transfers`      | 8190       |
-| `get_account_transfers` | 8190       |
-| `get_account_history`   | 8190       |
+| Operation               | Batch Size (Bytes) |
```

Comment:
This table counts events, not bytes.

Also, maybe we should split up the "request batch" and "reply batch".
`get_account_transfers` and `get_account_history` both take exactly 1 `AccountFilter`, but return (potentially) many more results.